lib/std/zig/system.zig

```diff
@@ -89,7 +89,13 @@ pub const NativePaths = struct {
         }
 
         if (!is_windows) {
-            const triple = try Target.current.linuxTriple(allocator);
+            // on linux the native abi might be different if we are statically linked
+            const target = if (Target.current.os.tag == .linux)
+                (try NativeTargetInfo.detectAbiAndDynamicLinker(allocator, Target.current.cpu, Target.current.os, .{})).target
```

Comment:
This function should have the target passed to it, because in practice, the calling code is also going to call `detectAbiAndDynamicLinker` to obtain a target, so this will effectively cause the detection to happen twice.